completions/tmux

```diff
@@ -0,0 +1,311 @@
+# tmux(1) completion                                       -*- shell-script -*-
+# SPDX-License-Identifier: GPL-2.0-or-later OR ISC
+
+# Log a message to help with debugging.
+# If BASH_COMPLETION_DEBUG is set, it will be printed to stderr.
+# When running with `set -x`, the _comp_cmd_tmux__log call itself will be
+# printed.
+#
+# @param $1 Message to log
+_comp_cmd_tmux__log()
+{
+    if [[ ${BASH_COMPLETION_DEBUG-} ]]; then
+        printf 'tmux bash completion: %s\n' "$1" >&2
+    fi
+}
+
+# Run the tmux command being completed.
+#
+# @param $@ args to tmux
+_comp_cmd_tmux__run()
+{
+    local -a REPLY
+    _comp_dequote "${comp_args[0]-}" || REPLY=("${comp_args[0]-}")
+    LC_ALL=C "${REPLY[@]}" "$@"
```

Comment:
Even if the command name produces multiple words (which happens when pathname expansions produce multiple filenames, etc.), what we want here is the command name, so I think we should use the first word. For example, even if `tmux*` would produce `tmux tmux1 tmux2`, what we want to execute is `tmux --help` instead of `tmux tmux1 tmux2 --help`.

```suggestion
    LC_ALL=C "${REPLY[0]}" "$@"
```

Reply:
Done. I was thinking that with something like the below it would make sense to include all words, but bash didn't seem to use the custom completion for that.

```
$ tmux='tmux -L foo'
$ $tmux <Tab>
```

Reply:
The user might register `complete -F _comp_cmd_tmux '$tmux'`. I also have an unsubmitted branch [`akinomyoga:__load_completion-dequote`](https://github.com/akinomyoga/bash-completion/tree/__load_completion-dequote) that includes commit 7ed6f6c479f80de7c821386c71a2f2b4f6da35e7, which attempts to enable completions for the command line `$tmux`, etc. In that case, I think we want to run `tmux ...` to generate completions (instead of `tmux -L foo ...`).